2019/pt-br/src/0xa8-injection.md

```diff
@@ -0,0 +1,88 @@
+API8:2019 Injeção
+=================
+
+| Agentes/Vetores | Fraquezas de Segurança | Impactos |
+| - | - | - |
+| Específico da API : Explorabilidade **3** | Prevalência **2** : Detecção **3** | Técnico **3** : Específico do negócio |
+| Atacantes vão entregar à API dados maliciosos a quaisquer vetores de injeção disponíveis (ex.: dados, parâmetros, integrações e etc.), esperando que estes dados sejam entregues a um interpretador. | Falhas por injeção são muito comuns e geralmente encontrados em consultas SQL, LDAP ou noSQL, comando em SO, *parsers* de XML e ORM. São falhar simples de encontrar ao revisar código fonte, e atacantes podem utilizar *scanners* e *fuzzers*. | Injeção pode levar ao vazamento de informação e perda de dados. Também podem levar à DoS ou a perda completa de um *host*. |
+
+## A API está vulnerável?
+
+A API está vulnerável às falhas de injeção se:
+
+* Dados enviados pelo cliente não são validados, filtrados ou sanitizados pela API.
+* Dados enviados pelo cliente são utilizados diretamente ou concatenados para consultas SQL/NoSQL/LDAP, comandos de sistema operacional, *parsers* XML, ORM (*Object Relational Mapping*) ou ODM (*Object Document Mapper*).
+* Dados vindos de sistemas externos (ex.: sistemas de integração) não são validados, filtrados ou sanitizados pela API.
+
+## Cenários de exemplo de ataques
+
+### Cenário #1
+
+O *firmware* de um dispositivo de controle parental provê o *endpoint* `/api/CONFIG/restore` o qual espera um appId a ser enviado como um parâmetro *multipart*. Utilizando um descompilador, o atacante encontra que o parâmetro appId é repassado diretamente para uma chamada de sistema sem qualquer sanitização:
+
+```c
+snprintf(cmd, 128, "%srestore_backup.sh /tmp/postfile.bin %s %d",
+         "/mnt/shares/usr/bin/scripts/", appid, 66);
+system(cmd);
+```
+Com o seguinte comando o atacante consegue desligar qualquer dispositivo que estiverem com o *firmware* vulnerável:
```

Comment:
fix
* where it's written "_(...) qualquer dispositivo que **estiverem** com o *firmware* vulnerável (...)_"
* it should be "_(...) qualquer dispositivo que **estiver** com o *firmware* vulnerável (...)_"